src/gt4py/stencil_builder.py

```diff
@@ -125,8 +123,10 @@ def with_caching(
         -----
         Resets all cached build data.
         """
-        self._build_data = {}
-        self.caching = gt4py.caching.strategy_factory(caching_strategy_name, self, *args, **kwargs)
+        self._build_data: Dict[str, Any] = {}
+        self.caching = gt4py.caching.strategy_factory(
+            caching_strategy_name, self, *args, **self.options.cache_settings, **kwargs
+        )
```

Comment:
To avoid duplicated arguments, I'd do e.g.
```suggestion
        kwargs = self.options.cache_settings.copy().update(kwargs)
        self.caching = gt4py.caching.strategy_factory(
            caching_strategy_name, self, *args, **kwargs
        )
```

Reply:
Good catch! I have opted for this version:
```
kwargs = {**self.options.cache_settings, **kwargs}
```